src/main/java/mate/academy/spring/boot/controller/BookController.java

```diff
@@ -27,29 +31,43 @@ public BookController(BookService bookService) {
         this.bookService = bookService;
     }
 
-    @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
-    }
-
-    @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
-    public void delete(@PathVariable Long id) {
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(summary = "Delete the book", description = "Delete the book by ID")
+    public void deleteBookById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get books", description = "Get the book by ID")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.createBook(bookDto);
     }
 
     @GetMapping("/search")
-    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
-        return bookService.search(searchParameters);
+    @Operation(
+            summary = "Search for books",
+            description = "Retrieve a list of books based on the provided search parameters."
+    )
+    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, Pageable pageable) {
+        return bookService.search(searchParameters, pageable);
+    }
+
+    @GetMapping("/books")
+    @Operation(summary = "Get all books", description = "Get the all book")
+    public List<BookDto> getAllBooks() {
```

Comment:
What the difference between getAllBooks and findAll? 
let's leave List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable)